sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -379,17 +418,44 @@ impl TransportWrapperBuilder {
         }
         let spi_conf_map = Self::consolidate_spi_conf_map(&self.spi_map, &self.spi_conf_list)?;
         let i2c_conf_map = Self::consolidate_i2c_conf_map(&self.i2c_map, &self.i2c_conf_list)?;
-        Ok(TransportWrapper {
-            transport,
+        let mut transport_wrapper = TransportWrapper {
+            transport: Rc::from(transport),
             pin_map: self.pin_alias_map,
+            artificial_pin_map: HashMap::new(),
             uart_map: self.uart_map,
             spi_map: self.spi_map,
             i2c_map: self.i2c_map,
             pin_conf_map,
             spi_conf_map,
             i2c_conf_map,
             strapping_conf_map,
-        })
+        };
+        let mut io_expanders: HashMap<String, IoExpander> = HashMap::new();
+        for (name, conf) in self.io_expander_conf_map {
+            io_expanders.insert(
+                name.to_string(),
+                ioexpander::create(&conf, &transport_wrapper)?,
+            );
+        }
+        transport_wrapper
+            .artificial_pin_map
+            .insert("NULL".to_string(), Rc::new(NullPin::new()));
+        for (pinname, v) in self.pin_on_io_expander_map {
+            if let Some(io) = io_expanders.get(&v.io_expander) {
+                ensure!(
+                    (v.pin_no as usize) < io.pins.len(),
+                    TransportError::InvalidIoExpanderPinNo(v.io_expander.to_string(), v.pin_no)
+                );
+                transport_wrapper
+                    .artificial_pin_map
+                    .insert(pinname, io.pins[v.pin_no as usize].clone());
```

Comment:
The type of each element in `io.pins[...]` is `Rc<dyn GpioPin>` correct?  This should be `Rc::clone(&io.pins[...])`.

Reply:
Is there a difference in functionality, or is it just convention to explicitly write `Rc::` so that casual readers do not mistakenly think that we are cloning a complex object?

Reply:
Using the more explicit syntax is a rust convention for reference-counted pointer types.  It is a signal that you are cloning the reference, not performing a deep-clone of the contained object.
https://doc.rust-lang.org/book/ch15-04-rc.html
https://stackoverflow.com/questions/61949769/is-there-any-difference-between-rcclonerc-and-rc-clone-in-rust-is-there